src/reanimated2/layoutReanimation/web/componentUtils.ts

```diff
@@ -239,6 +281,23 @@ export function handleExitingAnimation(
   parent?.appendChild(dummy);
 
   const snapshot = snapshots.get(element)!;
+
+  const scrollOffsets = getElementScrollValue(element);
+
+  if (
+    scrollOffsets.scrollTopOffset !== snapshot.scrollOffsets.scrollTopOffset
+  ) {
+    snapshot.top +=
+      snapshot.scrollOffsets.scrollTopOffset - scrollOffsets.scrollTopOffset;
+  }
```

Comment:
The line `scrollOffsets.scrollTopOffset !== snapshot.scrollOffsets.scrollTopOffset` is a bit complicated. Let's use new variables:
```
const currentScrollTopOffset = scrollOffsets.scrollTopOffset;
const lastScrollTopOffset = snapshot.scrollOffsets.scrollTopOffset;
```
just in case you can add comment why `currentScrollTopOffset` can be different with `lastScrollTopOffset`, because this part potentially can be unintuitive for someone in the future.

Reply:
Done in [9278f1c](https://github.com/software-mansion/react-native-reanimated/pull/5626/commits/9278f1c6ea28f6cf5925e00728c55a74a8c1468d)